api/src/db/DynamoBusinessesDataClient.ts

```diff
@@ -0,0 +1,151 @@
+): BusinessesDataClient => {
+  const doMigration = async (data: any): Promise<Business> => {
+    const migratedData = migrateData(data);
+    await put(migratedData);
+    return migratedData;
+  };
+
+  const findByIndex = (
+    indexName: string,
+    keyConditionExpression: string,
+    expressionAttributeValues: { [key: string]: any }
+  ): Promise<Business | undefined> => {
+    const params: QueryCommandInput = {
+      TableName: tableName,
+      IndexName: indexName,
+      KeyConditionExpression: keyConditionExpression,
+      ExpressionAttributeValues: expressionAttributeValues,
+    };
+
+    return db
+      .send(new QueryCommand(params))
+      .then((result) => {
+        if (!result.Items || result.Items.length !== 1) {
+          return;
+        }
+        const item = unmarshall(result.Items[0]);
+        return doMigration(item.data);
+      })
+      .catch((error) => {
+        console.log(error);
+        throw new Error("Not found");
+      });
+  };
+
+  const findByNAICSCode = (naicsCode: string): Promise<Business | undefined> => {
+    return findByIndex("NaicsCode", "naicsCode = :naicsCode", { ":naicsCode": { S: naicsCode } });
+  };
+
+  const findByIndustry = (industry: string): Promise<Business | undefined> => {
+    return findByIndex("Industry", "industry = :industry", { ":industry": { S: industry } });
+  };
+
+  const findByBusinessName = (businessName: string): Promise<Business | undefined> => {
+    return findByIndex("BusinessName", "businessName = :businessName", {
+      ":businessName": { S: businessName },
+    });
+  };
+
+  const findByEncryptedTaxId = (encryptedTaxId: string): Promise<Business | undefined> => {
+    return findByIndex("EncryptedTaxId", "encryptedTaxId = :encryptedTaxId", {
+      ":encryptedTaxId": { S: encryptedTaxId },
+    });
+  };
+
+  const get = (businessId: string): Promise<Business> => {
+    const params = {
+      TableName: tableName,
+      Key: {
+        businessId: businessId,
+      },
+    };
+    return db
+      .send(new GetCommand(params))
+
+      .then((result) => {
+        if (!result.Item) {
+          throw new Error("Not found");
+        }
+        return doMigration(result.Item.data);
+      })
+      .catch((error) => {
+        throw error;
+      });
+  };
+
+  const put = async (businessData: Business): Promise<Business> => {
+    const migratedData = migrateData(businessData);
+    const params = {
+      TableName: tableName,
+      Item: {
+        businessId: migratedData.id,
+        businessName: migratedData.profileData.businessName,
+        naicsCode: migratedData.profileData.naicsCode,
+        industry: migratedData.profileData.industryId,
+        encryptedTaxId: migratedData.profileData.encryptedTaxId,
+        data: migratedData,
+      },
+    };
+
+    return db
+      .send(new PutCommand(params))
+      .then(() => {
+        return migratedData;
+      })
+      .catch((error) => {
+        throw error;
+      });
```

Comment:
[dust] Nitpick but I think this is redundant. It's in at least one other place as well.

Reply:
Good catch!,  I've removed the redundant catch block here and in the other places as well. Thanks for pointing that out!